ddtrace/appsec/_iast/sampling/vulnerability_detection.py

```diff
@@ -0,0 +1,131 @@
+import collections
+from typing import Dict
+
+from ddtrace.appsec._iast._iast_env import _get_iast_env
+from ddtrace.internal.logger import get_logger
+from ddtrace.settings.asm import config as asm_config
+
+
+log = get_logger(__name__)
+
+# Global vulnerabilities map per endpoint - Map<String, Map<VulnerabilityType, int>>
+# Key is a string combining the request method (GET, POST, ...) and HTTP route (endpoint)
+# Value is another Map where the key is vulnerability type, and value is count
+# Using LRU cache to limit memory usage
+MAX_ENDPOINTS = 4096
+
+RequestDictType = Dict[str, int]
+GlobalDictType = Dict[str, RequestDictType]
+GLOBAL_VULNERABILITIES_LIMIT: GlobalDictType = collections.OrderedDict()
+
+
+def _get_global_limit():
+    global GLOBAL_VULNERABILITIES_LIMIT
+    return GLOBAL_VULNERABILITIES_LIMIT
+
+
+def _reset_global_limit():
+    global GLOBAL_VULNERABILITIES_LIMIT
+    GLOBAL_VULNERABILITIES_LIMIT = collections.OrderedDict()
+
+
+def init_request_vulnerability_maps(context) -> RequestDictType:
+    """
+    Initialize vulnerability maps for a new request.
+
+    Creates a copy of the stored counts from the global map and a new empty map for
+    tracking vulnerabilities in the current request.
+    """
+    gm = _get_global_limit()
+
+    # Get the stored vulnerability counts for this endpoint, or empty dict if not found
+    if context.endpoint_key in gm:
+        # Move the accessed key to the end to implement LRU behavior
+        stored_counts: RequestDictType = gm.get(context.endpoint_key)
+        # Create a copy to use as reference throughout the request
+        copy_global_dict = stored_counts.copy()
+    else:
+        copy_global_dict = {}
+
+    return copy_global_dict
```

Comment:
or just (which will be faster)
return gm.get(context.endpoint_key, {}).copy()

Reply:
so with a single line function it becomes
`return GLOBAL_VULNERABILITIES_LIMIT.get(context.endpoint_key, {}).copy()`